Cron/KlSyncs.php

```diff
@@ -124,29 +126,37 @@ private function sendUpdatesToApp(array $groupedRows, bool $isRetry = false)
 
             if (in_array($topic, $trackApiTopics) && !empty($rows)) {
                 foreach($rows as $row) {
-                    $decodedPayload = json_decode($row['payload'], true);
-
-                    $eventTime = $decodedPayload['time'];
-                    unset($decodedPayload['time']);
-
-                    //TODO: if conditional for backward compatibility, needs to be removed in future versions
-                    $storeId = '';
-                    if (isset($decodedPayload['StoreId']))
-                    {
-                        $storeId = $decodedPayload['StoreId'];
-                        unset($decodedPayload['StoreId']);
+                    try {
+                        $decodedPayload = json_decode($row['payload'], true);
+
+                        $eventTime = $decodedPayload['time'];
+                        unset($decodedPayload['time']);
+
+                        //TODO: if conditional for backward compatibility, needs to be removed in future versions
+                        $storeId = '';
+                        if (isset($decodedPayload['StoreId']))
+                        {
+                            $storeId = $decodedPayload['StoreId'];
+                            unset($decodedPayload['StoreId']);
+                        }
+
+                        $response = $this->_dataHelper->klaviyoTrackEvent(
+                            $row['topic'],
+                            json_decode($row['user_properties'], true ),
+                            $decodedPayload,
+                            $eventTime,
+                            $storeId
+                        );
+                        if (!$response) {$response = '0';}
+
+                        array_push($responseManifest["$response"], $row['id']);
+                    } catch ( \Exception $e) {
+                        // the payload was likely truncated, this will catch any indexing errors
+                        // defaults to a failed response and allows the other rows to continue syncing
```

Comment:
Can we update this to reflect why this might fail if we're not trying to send truncated payloads in the first place?